Recruit-Training/Advanced-Recruit-Training/Telemetry-Teaser/src/App.tsx

```diff
@@ -2,24 +2,66 @@ import BatteryInput from "~/components/batteryInput";
 import Header from "~/components/header";
 import SpeedInput from "~/components/speedInput";
 import WeatherInput from "~/components/weatherInput";
+import Button from "~/components/Button/Button.tsx";
+import {useState} from "react";
 
 const App = () => {
-  const calculateRange = () => {
-    return;
+  const [speed,setSpeed] = useState(null);
+  const [batteryPercent, changeBatteryPercent] = useState(0);
+  const [weather, weatherChange] = useState(0);
+  const[speedValid,setSpeedValid]= useState(false);
+  const[batteryValid,setBatteryValid]= useState(false);
+  const [calc, setCalc] = useState(0)
+  const [initialCalc, setInitialCalc] =useState(false);
+
+  const calculateRange = (e) => {
+    e.preventDefault();
+    setCalc(-(speed * speed * batteryPercent / 2500) + (4 * batteryPercent) + weather);
+    setInitialCalc(true);
+  }
+  const speedChangeHandler = (e) => {
+    if(e.target.value > 90 || e.target.value < 0){
+      setSpeedValid(false)
+    }
+    else{
+      setSpeedValid(true)
+      setSpeed(e.target.value)
+      console.log(speed)
+    }
+}
+  const weatherChangeHandler = (e) => {
+    weatherChange(e.target.value);
+    console.log(weather)
   }
+const batterChangeHandler = (e) => {
+    if(e.target.value > 100 || e.target.value<0) {
```

Comment:
We should be validating if the input is a number or not, I was able to type in "e" and "-" into the input box. This is because the number input can accept floating-point numbers (e)